### PR TITLE
[SCB-2639]Support microservice component labels and instance routing labels register to service center

### PR DESCRIPTION
**Pull request by Vientu**
…check in to the registry.

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install -Pit` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: huawei-cloud/environment/src/main/resources/mapping.yaml [hunk 1]
```diff
@@ -36,4 +36,8 @@ PAAS_PROJECT_NAME:
 #CAS_COMPONENT_NAME:
 #  - servicecomb.service.name
 #CAS_INSTANCE_VERSION:
-#  - servicecomb.service.version
\ No newline at end of file
+#  - servicecomb.service.version
+#SERVICECOMB_SERVICE_PROPS:
+#  - servicecomb.service.properties
+#SERVICECOMB_INSTANCE_PROPS:
+#  - servicecomb.instance.properties
```
Comment (liubao68) [inline huawei-cloud/environment/src/main/resources/mapping.yaml:43]: add EOL for this file

File: huawei-cloud/servicestage/src/main/java/org/apache/servicecomb/huaweicloud/servicestage/CasEnvConfig.java [hunk 2]
```diff
@@ -57,10 +62,31 @@ private void init() {
         .getInstance().getStringProperty(INSTANCE_ID, EMPTY).get());
     properties.put(ENVIRONMENT_ID, DynamicPropertyFactory
         .getInstance().getStringProperty(ENVIRONMENT_ID, EMPTY).get());
+    properties.put(SERVICE_PROPS, DynamicPropertyFactory
+            .getInstance().getStringProperty(SERVICE_PROPS, EMPTY).get());
+    properties.put(INSTANCE_PROPS, DynamicPropertyFactory
+            .getInstance().getStringProperty(INSTANCE_PROPS, EMPTY).get());
```
Comment (liubao68) [inline huawei-cloud/servicestage/src/main/java/org/apache/servicecomb/huaweicloud/servicestage/CasEnvConfig.java:68]: Maybe read this 2 properties in getNonEmptyInstanceProperties and getNonEmptyServiceProperties is better.

File: huawei-cloud/servicestage/src/main/java/org/apache/servicecomb/huaweicloud/servicestage/CasEnvConfig.java [hunk 3]
```diff
@@ -57,10 +62,31 @@ private void init() {
...
+  }
+
+  private Map<String, String> parseProps(String value) {
+    Map<String, String> rs = new HashMap<>();
+    if (StringUtils.isEmpty(value)) {
+      return rs;
+    }
+    return Arrays.stream(value.split(",")).map(v -> v.split(":"))
+            .collect(Collectors.toMap(v -> v[0], v -> v[1]));
```
Comment (liubao68) [inline huawei-cloud/servicestage/src/main/java/org/apache/servicecomb/huaweicloud/servicestage/CasEnvConfig.java:77]: better check v size is 2.

File: huawei-cloud/servicestage/src/main/java/org/apache/servicecomb/huaweicloud/servicestage/CasEnvConfig.java [hunk 4]
```diff
@@ -57,10 +62,31 @@ private void init() {
...
+  }
+
+  public Map<String, String> getNonEmptyInstanceProperties() {
+    properties.putAll(parseProps(properties.get(INSTANCE_PROPS)));
```
Comment (liubao68) [inline huawei-cloud/servicestage/src/main/java/org/apache/servicecomb/huaweicloud/servicestage/CasEnvConfig.java:81]: Do not change properties is better.

File: huawei-cloud/servicestage/src/main/java/org/apache/servicecomb/huaweicloud/servicestage/CasEnvConfig.java (hunk 2 above)
Comment (Vientu) [reply huawei-cloud/servicestage/src/main/java/org/apache/servicecomb/huaweicloud/servicestage/CasEnvConfig.java]: OK

File: huawei-cloud/servicestage/src/main/java/org/apache/servicecomb/huaweicloud/servicestage/CasEnvConfig.java (hunk 3 above)
Comment (Vientu) [reply huawei-cloud/servicestage/src/main/java/org/apache/servicecomb/huaweicloud/servicestage/CasEnvConfig.java]: OK

File: huawei-cloud/servicestage/src/main/java/org/apache/servicecomb/huaweicloud/servicestage/CasEnvConfig.java (hunk 4 above)
Comment (Vientu) [reply huawei-cloud/servicestage/src/main/java/org/apache/servicecomb/huaweicloud/servicestage/CasEnvConfig.java]: OK

**Review by liubao68 — APPROVED** [review]

**Pull request merged**